### ISSUE TITLE
Needs a Makefile

### ISSUE THREAD

**Issue by secrettriangle** [seenaburns/dex-ui#2]

**Comment by reanimus** [comment]
Not really, just modify one of the openframeworks ones.

This is the one I use (literally copied off of the example one)

```
# Attempt to load a config.make file.
# If none is found, project defaults in config.project.make will be used.
ifneq ($(wildcard config.make),)
	include config.make
endif

# make sure the the OF_ROOT location is defined
ifndef OF_ROOT
    OF_ROOT=../../..
endif

# call the project makefile!
include $(OF_ROOT)/libs/openFrameworksCompiled/project/makefileCommon/compile.project.mk
```

**Comment by reanimus** [comment]
You will, of course, need to provide a correct OF_ROOT

**Comment by cengbrecht** [comment]
So being new to this method, could you explain how to define the root, and the project makefile? 

I tried setting OF_ROOT to ./src but then it complained about the include?

**Comment by anthonyclays** [comment]
You have to set OF_ROOT to the root directory of your openframeworks installation. If you downloaded openframeworks to ~/Downloads, this should be OF_ROOT=~/Downloads/openFrameworks

**Comment by cengbrecht** [comment]
Being new still, I do not know where to get openFrameworks?

**Comment by anthonyclays** [comment]
Download openframeworks from [here](http://openframeworks.cc/). Choose your operating system and follow the instructions. When everything is set up correctly, create a standard Makefile (either copy from the openframeworks examples or use the project generator that ships with it). You'll be able to compile this project (though it will not yet feel as polished as in the video). If you need anymore help, feel free to send me a message or ask help on an openframeworks-related IRC channel.

About the actual issue: I believe it'd be a good idea to include a standard Makefile in the repository, so that's one vote in favor.

**Comment by cengbrecht** [comment]
Two votes here. :-P

**Comment by CestDiego** [comment]
I don't know if my vote counts, but ... three votes :D

**Comment by RMKD** [comment]
agreed. this seems like a cool project but the README does not provide any installation guidelines. anyone have a step-by-step if you've never used Open Frameworks?

**Comment by ryanmjacobs** [comment]
@RMKD I literally just got this working.

Step 1. Compile openFrameworks

```
$ cd /tmp
$ git clone --depth=1 https://github.com/openframeworks/openFrameworks
$ cd openFrameworks/scripts/linux/
$ sudo ./<your_distro>/install_dependencies.sh
$ ./compileOF.sh
```

Step 2. Clone Dex UI

```
$ cd /tmp
$ git clone --depth=1 https://github.com/seenaburns/dex-ui
```

Step 3. Copy @reanimus's Makefile into "/tmp/dex-ui/Makefile"

Step 4. Compile Dex UI

```
$ cd /tmp/dex-ui
$ sed -i '1i OF_ROOT=/tmp/openFrameworks' Makefile
$ make
```

Step 5. Run Dex UI

```
$ cd /tmp/dex-ui/bin
$ ./dex-ui
```

**Comment by RMKD** [comment]
thx @ryanmjacobs - I'm compiling now but getting errors on run. If it's helpful to anyone, I got rid of a bunch of image and font errors by (1) adding placeholder dummy images /tmp/dex-ui/bin/data/images/ref-img.png and /tmp/dex-ui/bin/data/images/noise-bg-color-darker.png and (2) editing src/graphics-utils.cpp to use DroidSans rather than an empty string for the font. Still getting a seg-fault on run. Was there anything else people have been adding/editing to run this?

**Comment by trengrj** [comment]
@RMKD I managed to fix the Seg faulting but running with `sudo`. It just quits with no errors but I haven't setup bspwm yet.

**Comment by wsturgiss** [comment]
I'm trying to get this running on a mac.  I have openFrameworks working via xcode, but I have yet to figure out how to compile it via the command line.  When I look in the osx folder in the scripts folder, there is no compile file, so I'm not sure what to do at this point.  I thought building it in xcode might compile it, so I did that and then moved on, but then I got tons of error when it came time to compile Dex ui.  If anyone ever gets this running on a mac, I'd love to know how they did.
